### DISCUSSION TITLE
Hide "Includes paid promotion"

### DISCUSSION THREAD

**Discussion by emilypeto** [ReVanced/revanced-discussions#187] [Question] [answered]
This notification appears at the beginning of many videos:
![Screenshot_20220511-122657_YouTube](https://user-images.githubusercontent.com/46334960/167930084-bf78738b-64f9-4c30-a2f7-faebe5711c2b.jpg)
It sticks around far too long and it's kind of annoying so an option to disable it would be nice. Kind of useless anyways since we have sponsorblock. I'm not sure if this only occurs in specific regions.

**Comment by oSumAtrIX** [comment]
Maby make it shorter or disable it completely

**Comment by MulverineX** [comment]
Perhaps this should be in the Patches section?

**Comment by emilypeto** [comment]
I'm working on implementing this now

**Reply by emilypeto** [reply]
In the constructor of com.google.android.apps.youtube.app.player.overlay.PipPaidProductBadgeOverlay there is a constant 10000 (0x2710), I believe this is the number of milliseconds to display the overlay. I'm having some trouble with actually writing the patch as I'm not very familiar with Java.

**Reply by emilypeto** [reply]
If anyone else wants to go ahead and write the actual patch for that value I think it would be great to replace it with a number*1000 pulled from a settings entry labeled "number of seconds". An easy way to change the constant would be to simply add some instructions at the end of the constructor method, setting the variable again to our own value.

**Reply by oSumAtrIX** [reply]
Have you tried to manually patch the constant and checking if it changes the display time

**Reply by emilypeto** [reply]
Having issues with apktool as well so I haven't been able to, but PipPaidProductBadgeOverlay is the name of the relevant item and I've timed that it displays for 10 seconds.

**Reply by emilypeto** [reply]
![Screenshot_20220531-091121_YouTube](https://user-images.githubusercontent.com/46334960/171220447-8ff8da51-127e-449b-a7bb-d48fbc1c3694.jpg)

**Reply by emilypeto** [reply]
If not there then there's also some relevant code at com.google.android.apps.youtube.embeddedplayer.service.ui.overlays.paidcontentoverlay, but the location named above seems to be the only place where a 10 second constant is defined

**Reply by emilypeto** [reply]
Upon further digging that same value gets set in a loose class as well, not sure if that will interfere. The class can be found with `egrep -ril 'paid.+?overlay' | xargs grep -n "0x2710"`

**Reply by oSumAtrIX** [reply]
Please investigate more into it, I can later create a patch for it if you can confirm that this is what needs to be changed

**Reply by j4k0xb** [reply]
I did some hooking, its a litho component 🤢 (so it cant be in `PipPaidProductBadgeOverlay` or `zav`)
Welcome in the boat of postponed patches because no one figured out how to deal with them yet

Proof of concept (YT v17.14.35, logs `dpl@73b8ff3 Includes paid promotion`):
```sh
frida -U -l hook.js -n YouTube
```
```js
Java.perform(() => {
  Java.choose('dpl', {
    onMatch(instance) {
      try {
        const text = instance._a.value._Q.value.toString();
        if (text.includes('paid')) console.log(instance, text);
      } catch (err) { }
    },
    onComplete() { },
  });
});
```

**Comment by JoshCode** [comment]
Please do not make hiding this default behaviour, disclaimers like these are vital to informing viewers that if they're about to listen to an opinion, that opinion was influenced by some form of sponsorship.

**Reply by emilypeto** [reply]
The idea now is to make the duration configurable

**Comment by oSumAtrIX** [comment, marked as answer]
Added with commit [a27ec45](https://github.com/revanced/revanced-integrations/commit/a27ec45be8c1d9916e41765f2bea3491d6805af6). Promotions overlay is now hidden by default by the integrations.
